import/index_java/src/org/vufind/index/ConfigManager.java

```diff
@@ -210,55 +221,59 @@ public Ini loadConfigFile(String filename)
                 retVal.put(key, overrideSection.get(key));
             }
         }
+
+        if (retVal == null) {
+            logger.warn(section + " section missing from " + filename);
```

Comment:
Since this code is now getting used more broadly (because of refactoring of the "get setting" methods and changes to default behavior), I think this log message is now going to be problematic for the same reasons as the other log message I mentioned. Perhaps we need to consider the same options I suggested there: adding an argument, adding a method for fetching "required" sections, or a third option I didn't propose before: doing the check and log after calling the method rather than inside the method. What do you think?

Reply:
Not sure what should happen in these cases. If a setting or section really is "required", then it feels like logging a warning isn't enough and there should be an error or even exception instead.
And also, I don't think it would be a good idea to have long function names like getRawRequiredSanitizedConfigSection()... another option would be to have a global flag to define whether these warnings should be logged or not, or to set them to debug and control that over the log level. But i'm also not sure what the best solution would be here...

Reply:
I think my inclination is to drop the logging entirely. If something is required, the calling code should check the return value. If it's not required, having this log message is not helpful. The only place where this message would have been displayed before would have been from the calls you refactored in PunctuationContainer.java... so maybe we should log messages in there if getConfigSection returns empty values. Or maybe we don't even need to do that.

Reply:
Alright, so i removed the logging from the ConfigManager and added some checks to the PunctuationContainer. Hope it looks better now!